crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -90,6 +90,16 @@ pub enum NamedChannelOrUrl {
     Path(Utf8TypedPathBuf),
 }
 
+impl std::hash::Hash for NamedChannelOrUrl {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        match self {
+            NamedChannelOrUrl::Name(name) => name.hash(state),
+            NamedChannelOrUrl::Url(url) => url.as_str().hash(state),
+            NamedChannelOrUrl::Path(path) => path.as_str().hash(state),
+        }
+    }
+}
```

Comment:
Why was this needed?

Reply:
This was missing and got a compile error for the new rust version.

Reply:
But wasnt the derive good enough?

Reply:
I'm actually unsure how that derive could have ever worked because afaic `Utf8TypedPathBuf` does not derive Hash?

Reply:
It does implement Hash: https://docs.rs/typed-path/latest/typed_path/enum.Utf8TypedPathBuf.html#impl-Hash-for-Utf8TypedPathBuf

Reply:
```
error[E0277]: the trait bound `Utf8TypedPathBuf: Hash` is not satisfied
  --> crates/rattler_conda_types/src/channel/mod.rs:90:10
   |
81 | #[derive(Debug, Clone, Eq, PartialEq, Hash)]
   |                                       ---- in this derive macro expansion
...
90 |     Path(Utf8TypedPathBuf),
   |          ^^^^^^^^^^^^^^^^ the trait `Hash` is not implemented for `Utf8TypedPathBuf`
```

Looking at the docs, https://docs.rs/camino/latest/camino/struct.Utf8PathBuf.html#impl-Hash-for-Utf8PathBuf it seems it might do so now, we could also bump?

Reply:
Bumped to 0.9.2, 0.9.0 did not implement Hash.